pandas/core/groupby/ops.py

```diff
@@ -849,11 +848,18 @@ def agg_series(
 
         result = self._aggregate_series_pure_python(obj, func)
 
-        npvalues = lib.maybe_convert_objects(result, try_float=False)
-        if preserve_dtype:
-            out = maybe_cast_pointwise_result(npvalues, obj.dtype, numeric_only=True)
+        if len(obj) == 0 and len(result) == 0 and isinstance(obj.dtype, ExtensionDtype):
```

Comment:
What happens on an empty list of categoricals with `observed=False`? I think this is the only case where `len(obj) == 0` but `len(result) > 0`.

Reply:
I didn't track his down specifically, but the test that was mentioned in the comment is not passing by here anymore

Reply:
Running on this PR as-is:

```
func = "min"
dtype = "string[pyarrow_numpy]"
df = DataFrame({"a": ["a"], "b": "a", "c": "a"}, dtype=dtype)
df["a"] = pd.Categorical([1], categories=[1, 2, 3])
df = df.iloc[:0]
result = getattr(df.groupby("a", observed=False), func)()

print(result.dtypes)
# b    float64
# c    float64
# dtype: object
```

If you remove the condition `len(result) == 0` in the line highlighted above, you get `string[pyarrow_numpy]` for the dtypes instead. However, the values in the DataFrame are still the float NaN. I didn't realize this was possible?

Reply:
The new string dtype uses nan as missing value representation to keep numpy semantics, so it will work if you feed it an array with all NaN

Reply:
Ah - thanks

Reply:
I think this still needs fixed (but okay not for 2.1.2)